tests/ResponseTest.php

```diff
@@ -126,4 +185,97 @@ public function testIsType()
+    {
+        $response = new Response();
+
+        $this->assertInstanceOf(Response::class, $response->addHeader('Date', ['Sun', '18 Oct 2009 08:56:53 GMT']));
+        $this->assertSame('Sun, 18 Oct 2009 08:56:53 GMT', $response->getHeader('Date'));
+    }
+
+    public function testAddHeaderWithArrayValueAppendExistedHeader()
+    {
+        $response = new Response();
+        $response->addHeader('Date', 'Sun');
+        $response->addHeader('Date', ['18 Oct 2009 08:56:53 GMT']);
+
+        $this->assertSame('Sun, 18 Oct 2009 08:56:53 GMT', $response->getHeader('Date'));
+    }
+
+    public function testAddHeaders()
+    {
+        $response = new Response();
+        $result = $response->addHeaders([
+            'Content-Type' => 'text/html; charset=utf-8',
+            'Server' => 'Apache',
+        ]);
+
+        $this->assertInstanceOf(Response::class, $result);
+        $this->assertSame('text/html; charset=utf-8', $response->getHeader('Content-Type'));
+        $this->assertSame('Apache', $response->getHeader('Server'));
+    }
+
+    public function testGetHead()
+    {
+        $response = new Response();
+        $response->addHeaders([
+            'Content-Type' => 'text/html; charset=utf-8',
+            'Server' => 'Apache',
+        ]);
+        $expectedHead = 'HTTP/1.0 200 OK' . "\r\n" . 'Content-Type: text/html; charset=utf-8' . "\r\n" . 'Server: Apache' . "\r\n\r\n";
+
+        $this->assertSame($expectedHead, $response->getHead());
+    }
+
+    public function testGetContentLength()
+    {
+        $response = new Response();
+
+        $this->assertSame(0, $response->getContentLength());
+    }
+
+    public function testGetContentLengthShouldReturn19730()
+    {
+        $response = new Response();
+        $response->addHeader('Content-Length', 19730);
+
+        $this->assertSame(19730, $response->getContentLength());
+    }
+
+    public function testIsContentType()
+    {
+        $response = new Response();
+        $response->addHeader('Content-Type', 'text/html; charset=utf-8');
+
+        $this->assertTrue($response->isContentType('text/html'));
+    }
+
+    public function testClassInstanceShouldReturnMessageBodyString()
+    {
+        $response = new Response();
+        $response->setBody(new MessageBodyString('param1=value1&param2=value2'));
+        $response->addHeader('Content-Length', 19730);
+        $expectedString = 'HTTP/1.0 200 OK' . "\r\n" . 'Content-Length: 27' . "\r\n" . 'Content-Length: 19730' . "\r\n\r\n" . 'param1=value1&param2=value2';
+
+        $this->assertSame($expectedString, (string)$response);
+    }
+
+    public function testClassInstanceShouldBeCloned()
+    {
+        $response = new Response();
+        $response->setBody(new MessageBodyString('param1=value1&param2=value2'));
+        $cloneResponse = clone $response;
+
+        $this->assertInstanceOf(Response::class, $cloneResponse);
+        $this->assertInstanceOf(MessageBodyString::class, $cloneResponse->getBody());
```

Comment:
It would be nice to actually check a few values here, in particular the cloned body contents.